src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1242,8 +1472,6 @@ TxQ::accept(Application& app,
             requiredFeeLevel;
         if (feeLevelPaid >= requiredFeeLevel)
         {
-            auto firstTxn = candidateIter->txn;
-
```

Comment:
With the introduction of tickets "remove the last item of account" makes less sense than it used to.  But let's try to understand the motivation for the `dropPenalty`.

- If the idea is to simply open up another spot in the queue, then dropping an arbitrary ticket-transaction (actually the "last" one) is as good as dropping the last sequence-transaction.

- If the idea is to provide a punishment for a failing transaction in a nearly full queue then, again, dropping an arbitrary ticket-transaction (actually the "last" one) is as good as dropping the last sequence-transaction.

In either of those cases it seems like the _best_ option would be to drop the transaction with the lowest fee.  But we're in a rare circumstance, so I don't think it's worth the effort to search for the prefect transaction to remove.

Therefore I think, as long as we want to drop one of the account's queued transactions, dropping either the "last" ticket or the last sequence serves the purpose.  Therefore I'm not planning to change that part of the logic.  Please let me know if you disagree.  Thanks.

While we're looking at that block, I'm nervous about the
```
assert(endIter != candidateIter);
```
With tickets I'm no longer confident that will necessarily be the case.  My preference would be to replace the `assert` with
```
if (endIter != candidateIter)
    erase(endIter);
```
I feel like that's a bit more future proof.  I'll go ahead and make that change.  But please let me know if you disagree.  Thanks again.

Reply:
Great points. The idea is definitely to open up a spot in the queue for by dropping a less expensive transaction to allow a more expensive one in. Now that you mention it, I think the _bestest_ option would be to drop the transaction with the lowest fee from the set of { ticket-transactions, the last sequence-transaction }. I wouldn't want to drop a ticket from the middle of the sequences, because that would kill all the others after it. For tickets, that's not true.

However, all that said, simplicity has value, and given that this is a very rare circumstance, I am fine with leaving the drop as-is.

I definitely agree about the possibility that `endIter == candidateIter`, since Tickets are applied in fee order, and the last ticket by sequence order could easily be the one that pays most for this account. Your change looks good and safe.